src/setup/rgw.py

```diff
@@ -14,77 +14,221 @@
+                    rgw_lock = False
+                    break
+
+            except Exception as e:
+                Log.error('Exception occured while connecting to consul service'
+                    f' endpoint {e}')
+                break
+        if rgw_lock is True:
+            Log.info('Creating admin user.')
+            # TODO: Add rgw admin user creation.
+            Log.info('User is created.')
+            Log.info(f'Deleting rgw_lock key {rgw_lock_key}.')
+            Conf.delete(rgw_consul_idx, rgw_lock_key)
+            Log.info(f'{rgw_lock_key} key is deleted')
+
+        Log.info('Init phase completed.')
         return 0
 
     @staticmethod
-    def test(config_path: str, plan: str):
-        """
-        Perform configuration testing.
-        """
+    def test(cortx_gconf: MappedConf, plan: str):
+        """Perform configuration testing."""
 
+        Log.info('Test phase completed.')
         return 0
 
     @staticmethod
-    def reset(config_path: str):
-        """
-        Remove/Delete all the data/logs that was created by user/testing.
-        """
+    def reset(cortx_gconf: MappedConf):
+        """Remove/Delete all the data/logs that was created by user/testing."""
 
+        Log.info('Reset phase completed.')
         return 0
 
     @staticmethod
-    def cleanup(pre_factory: bool, config_path: str):
-        """
-        Remove/Delete all the data that was created after post install.
-        """
+    def cleanup(cortx_gconf: MappedConf, pre_factory: bool):
+        """Remove/Delete all the data that was created after post install."""
 
+        Log.info('Cleanup phase completed.')
         return 0
 
     @staticmethod
-    def upgrade(config_path: str):
-        """
-        Perform upgrade steps.
-        """
+    def upgrade(cortx_gconf: MappedConf):
+        """Perform upgrade steps."""
 
+        Log.info('Upgrade phase completed.')
         return 0
+
+    @staticmethod
+    def _get_consul_url(cortx_gconf: MappedConf):
+        """Return consul url."""
+
+        cortx_consul_endpoint_key = 'cortx>external>consul>endpoints'
+        consul_endpoint_list = cortx_gconf.get(cortx_consul_endpoint_key)
+        consul_fqdn = None
+        for endpoint in consul_endpoint_list:
+            if 'http' in endpoint:
+                consul_fqdn = endpoint.split('http')[1]
+                break
+        consul_url = 'consul' + consul_fqdn
+        return consul_url
+
+    @staticmethod
+    def _file_exist(file_path: str):
+        """Return True if given file exist else False."""
+        exist = True
+        if not os.path.exists(file_path):
+            exist = False
+            raise RgwSetupError(errno.EINVAL,
+                f'{file_path} file not exists.')
+        return exist
+
+    @staticmethod
+    def _update_config(conf_idx: str, kv_list: list, conf_url: str = None):
+        """Add/Updated key-values in given config."""
+        try:
+            if conf_url is not None:
```

Comment:
what if conf_url is None? Can we validate like below
```
 if conf_url is None:
     raise Error
 Conf.load(conf_idx, conf_url, skip_reload=True)
 ```

Reply:
If url is already loaded and we just want to add key-values that why I have added this check.
 To avoid confusion added 'load_conf' parameter. 
```
 if load_conf:
                if conf_url is None:
                    raise RgwSetupError(errno.EINVAL, 'Conf url is None.')
                Conf.load(conf_idx, conf_url, skip_reload=True)
```